### PR TITLE
feat: define appium home for plugins to handle multiple drivers

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

This issue is still WIP.

When I managed different versions' drivers and plugins on a host, I got a use case I wanted to use a particular driver version with plugins. To reduce installation time, we can have multiple `appium-home` on a host for drivers and can switch them when an appium server starts. 

For now, plugins could have multiple modules in a session, but drivers could be used only one module in a session.
So, like the below case could do.

```
appium --appium-home=/path/to/driver/home --appium-plugin-home=/path/to/plugin/home
```

```
appium --appium-home=/path/to/driver/which/has/another/version --appium-plugin-home=/path/to/plugin/home
```

This case may help to reduce installation time to manage multiple driver versions on a host and use the same plugin versions across them. This case may happen frequently than managing a plugin as various versions.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jlipps — COMMENTED** [review]
this looks interesting, is there a scenario where we would also want `--appium-driver-home`? also does using this new flag do the right thing in terms of creating an appium manifest file in the plugin home directory, etc?

**Comment by jlipps** [comment]
also we probably want to rebase this into the monorepo setup after that is merged

**Comment by KazuCocoa** [comment]
> --appium-driver-home

I considered the case, but probably only for plugins is enough for now.
As appium, drivers are mondatory to enable `automationName`s. So `--appium-home` can think this home is mainly for drivers. if we want to manage driver versions, e.g. for development and for production in our development case, we can define multiple `--appium-home` for each usage. Plugins are kindly an optional, so we can extract plugin version manageemnt outside drivers. 

> does using this new flag do the right thing in terms of creating an appium manifest file in the plugin home directory, etc?

Still considering.
Probably its better to split `drivers` and `plugins` from `extensions.yaml` for each directory so far.

**Comment by jlipps** [comment]
@KazuCocoa do you want to rebase this on the monorepo?

File: lib/cli/args.js [hunk 1]
```diff
@@ -24,6 +24,13 @@ const sharedArgs = [
     dest: 'appiumHome',
   }],
 
+  [['-aph', '--appium-plugin-home'], {
```
Comment (Dharin-shah) [inline lib/cli/args.js:27]: `--plugin-home` as an option also required i believe

**Comment by KazuCocoa** [comment]
will rebase & update this PR this weekend or around :)

**Comment by KazuCocoa** [comment]
=> https://github.com/appium/appium/pull/15500

**Pull request closed without merging**